internal/cmd/connect/install_utils.go

```diff
@@ -0,0 +1,439 @@
+package connect
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/confluentinc/properties"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/form"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/types"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+)
+
+type platformInstallation struct {
+	Location platformLocation
+	Use      string
+}
+
+type platformLocation struct {
+	Type string
+	Path string
+}
+
+type WorkerConfig struct {
+	Path string
+	Use  string
+}
+
+type listOut struct {
+	Number      string `human:""`
+	Path        string `human:"Path"`
+	Description string `human:"Description"`
+}
+
+func getConfluentPlatformInstallation(cmd *cobra.Command, force bool) (*platformInstallation, error) {
+	installations, err := findInstallationDirectories()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(installations) == 0 {
+		return nil, errors.NewErrorWithSuggestions("unable to detect a Confluent Platform installation", "Pass the plugin directory and worker configuration files with `--plugin-directory` and `--worker-configs`.")
+	} else if force {
+		output.Printf("\nUsing the Confluent Platform installation at \"%s\".\n", installations[0].Location.Path)
+		return &installations[0], nil
+	} else if len(installations) == 1 {
+		output.Printf("\nUsing the only available Confluent Platform installation at \"%s\".\n", installations[0].Location.Path)
+		return &installations[0], nil
+	} else {
+		list := output.NewList(cmd)
+		for i, installation := range installations {
+			list.Add(&listOut{
+				Number:      strconv.Itoa(i + 1),
+				Path:        installation.Location.Path,
+				Description: installation.Use,
+			})
+		}
+
+		listStr, err := list.PrintString()
+		if err != nil {
+			return nil, err
+		}
+
+		promptMsg := "\nThe plugin can be installed in any of the following Confluent Platform installations. Enter the number corresponding to the installation you would like to use:\n%sTo cancel, press Ctrl-C"
+		f := form.New(form.Field{
+			ID:     "installation",
+			Prompt: fmt.Sprintf(promptMsg, listStr),
+			Regex:  `^\d$`,
+		})
+		if err := f.Prompt(form.NewPrompt(os.Stdin)); err != nil {
+			return nil, err
+		}
+		choice, err := strconv.Atoi(f.Responses["installation"].(string))
+		if err != nil || choice < 1 || choice > len(installations) {
+			return nil, errors.Errorf("your choice must be in the range %d to %d (inclusive)", 1, len(installations))
+		}
+		return &installations[choice-1], nil
+	}
+}
+
+func findInstallationDirectories() ([]platformInstallation, error) {
```

Comment:
does these commands require more read/write access than our plugin commands before? like to certain paths?

Reply:
I'm not sure what read/write permissions the plugin commands need, but this command needs read access to for the various paths checked here (it looks for the existence of specific sub-directories to detect Confluent Platform installations), and it needs read/write permissions for installing the plugin and updating the worker config files.